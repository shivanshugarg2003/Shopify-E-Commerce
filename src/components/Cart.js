import React from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { removeFromCart, increaseQuantity, decreaseQuantity } from '../redux/cartSlice';

const Cart = () => {
  const cart = useSelector(state => state.cart);
  console.log('Cart State:', cart);
  const dispatch = useDispatch();

  const total = cart.reduce(
    (sum, item) => sum + item.price * item.quantity,
    0
  );

  return (
    <div style={{ padding: '2rem' }}>
      <h2>Your Cart</h2>
      {cart.length === 0 ? (
        <p>Your cart is empty.</p>
      ) : (
        <>
          {cart.map(item => (
            <div
              key={item.id}
              style={{
                border: '1px solid #ccc',
                borderRadius: '5px',
                padding: '1rem',
                marginBottom: '1rem',
              }}
            >
              <h4>{item.name}</h4>
              <p data-testid={`quantity-${item.id}`}>
                Quantity:
                <button
                  onClick={() => dispatch(decreaseQuantity(item.id))}
                  style={{ margin: '0 6px', padding: '4px 10px' }}
                >
                  −
                </button>
                {item.quantity}
                <button
                  onClick={() => dispatch(increaseQuantity(item.id))}
                  style={{ margin: '0 6px', padding: '4px 10px' }}
                >
                  +
                </button>
              </p>
              <p>Total: ₹{item.price * item.quantity}</p>
              <button
                onClick={() => dispatch(removeFromCart(item.id))}
                style={{
                  backgroundColor: '#dc3545',
                  color: '#fff',
                  border: 'none',
                  padding: '6px 12px',
                  borderRadius: '4px',
                  cursor: 'pointer',
                  marginTop: '0.5rem',
                }}
              >
                Remove
              </button>
            </div>
          ))}
          <hr />
          <h3>Total Amount: ₹{total}</h3>
          <button
            style={{
              backgroundColor: '#007bff',
              color: '#fff',
              padding: '10px 20px',
              border: 'none',
              borderRadius: '5px',
              cursor: 'pointer',
            }}
          >
            Proceed to Checkout
          </button>
        </>
      )}
    </div>
  );
};

export default Cart;