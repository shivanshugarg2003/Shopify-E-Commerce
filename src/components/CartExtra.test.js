import React from 'react'; // ✅ Missing import
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import Cart from './Cart';
import { configureStore } from '@reduxjs/toolkit';
import cartReducer from '../redux/cartSlice';
import userEvent from '@testing-library/user-event';

const store = configureStore({
  reducer: {
    cart: cartReducer,
  },
});

describe('Cart.js Extra Coverage', () => {
  const setup = () =>
    render(
      <Provider store={store}>
        <Cart />
      </Provider>
    );

  it('renders Cart header', () => {
    setup();
    expect(screen.getByRole('heading', { level: 2, name: /your cart/i })).toBeInTheDocument();
  });

  it('shows empty cart message when no items', () => {
    setup();
    expect(screen.getByText(/your cart is empty/i)).toBeInTheDocument();
  });

  it('renders cart with items and displays them correctly', () => {
    store.dispatch({
      type: 'cart/addToCart',
      payload: {
        id: 1,
        name: 'T-shirt',
        price: 1000,
        quantity: 2,
        image: 'https://example.com/tshirt.jpg'
      },
    });

    setup();

    expect(screen.getByText(/t-shirt/i)).toBeInTheDocument();
    const priceElements = screen.getAllByText(/₹1000/i);
expect(priceElements.length).toBeGreaterThanOrEqual(1);
  });

  it('matches snapshot with empty cart', () => {
    // Remove all items from the cart to ensure it is empty
    store.dispatch({ type: 'cart/removeFromCart', payload: 1 });
    store.dispatch({ type: 'cart/removeFromCart', payload: 2 });
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('matches snapshot with items in cart', () => {
    // Remove any prior items and add only Jeans
    store.dispatch({ type: 'cart/removeFromCart', payload: 1 });
    store.dispatch({ type: 'cart/removeFromCart', payload: 2 });
    store.dispatch({
      type: 'cart/addToCart',
      payload: {
        id: 2,
        name: 'Jeans',
        price: 2000,
        quantity: 1,
        image: 'https://example.com/jeans.jpg'
      },
    });

    const { container } = setup();
    expect(container).toMatchSnapshot();
  });
});