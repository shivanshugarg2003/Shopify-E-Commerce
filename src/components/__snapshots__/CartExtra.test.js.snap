// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart.js Extra Coverage matches snapshot with empty cart 1`] = `
<div>
  <div
    style="padding: 2rem;"
  >
    <h2>
      Your Cart
    </h2>
    <p>
      Your cart is empty.
    </p>
  </div>
</div>
`;

exports[`Cart.js Extra Coverage matches snapshot with items in cart 1`] = `
<div>
  <div
    style="padding: 2rem;"
  >
    <h2>
      Your Cart
    </h2>
    <div
      style="border: 1px solid #ccc; border-radius: 5px; padding: 1rem; margin-bottom: 1rem;"
    >
      <h4>
        Jeans
      </h4>
      <p
        data-testid="quantity-2"
      >
        Quantity:
        <button
          style="margin: 0px 6px; padding: 4px 10px;"
        >
          −
        </button>
        1
        <button
          style="margin: 0px 6px; padding: 4px 10px;"
        >
          +
        </button>
      </p>
      <p>
        Total: ₹
        2000
      </p>
      <button
        style="background-color: rgb(220, 53, 69); color: rgb(255, 255, 255); padding: 6px 12px; border-radius: 4px; cursor: pointer; margin-top: 0.5rem;"
      >
        Remove
      </button>
    </div>
    <hr />
    <h3>
      Total Amount: ₹
      2000
    </h3>
    <button
      style="background-color: rgb(0, 123, 255); color: rgb(255, 255, 255); padding: 10px 20px; border-radius: 5px; cursor: pointer;"
    >
      Proceed to Checkout
    </button>
  </div>
</div>
`;
